src/phoenix/trace/openai/instrumentor.py

```diff
@@ -0,0 +1,211 @@
+                INSTRUMENTED_ATTRIBUTE_NAME,
+                True,
+            )
+
+
+def _wrap_openai_api_requestor(
+    request_fn: Callable[..., Any], tracer: Tracer
+) -> Callable[..., Any]:
+    """Wraps the OpenAI APIRequestor.request method to create spans for each API call.
+
+    Args:
+        request_fn (Callable[..., Any]): The request method on openai.api_requestor.APIRequestor.
+        tracer (Tracer): The tracer to use to create spans.
+
+    Returns:
+        Callable[..., Any]: The wrapped request method.
+    """
+
+    def wrapped(*args: Any, **kwargs: Any) -> Any:
+        call_signature = signature(request_fn)
+        bound_arguments = call_signature.bind(*args, **kwargs)
+        parameters = bound_arguments.arguments["params"]
+        url = bound_arguments.arguments["url"]
+        if _get_request_type(url) is RequestType.CHAT_COMPLETION:
+            current_status_code = SpanStatusCode.UNSET
+            start_time = datetime.datetime.now()
+            events: List[SpanEvent] = []
+            attributes: SpanAttributes = {}
+            attributes.update(_inputs(parameters))
+            attributes.update(_input_messages(parameters))
+            attributes.update(_invocation_parameters(parameters))
+            try:
+                outputs = request_fn(*args, **kwargs)
+                response = outputs[0]
+                attributes.update(_outputs(response))
+                attributes.update(_token_counts(response))
+                attributes.update(_function_calls(response))
+                current_status_code = SpanStatusCode.OK
+                return outputs
+            except Exception as error:
+                current_status_code = SpanStatusCode.ERROR
+                events.append(
+                    SpanException(
+                        message=str(error),
+                        timestamp=start_time,
+                        exception_type=type(error).__name__,
+                        exception_stacktrace=get_stacktrace(error),
+                    )
+                )
+                raise
+            finally:
+                tracer.create_span(
+                    name="openai.ChatCompletion.create",
+                    span_kind=SpanKind.LLM,
+                    start_time=start_time,
+                    end_time=datetime.datetime.now(),
+                    status_code=current_status_code,
+                    status_message="",
+                    attributes=attributes,
+                    events=events,
+                )
+        else:
+            return request_fn(*args, **kwargs)
+
+    return wrapped
+
+
+def _inputs(parameters: Dict[str, Any]) -> Iterator[Tuple[str, str]]:
+    """Yield input messages as a JSON string in addition to input mime type"""
+    if messages := parameters.get("messages"):
+        yield INPUT_VALUE, json.dumps([_get_openinference_message(message) for message in messages])
+        yield INPUT_MIME_TYPE, MimeType.JSON.value
```

Comment:
```suggestion
        yield INPUT_MIME_TYPE, MimeType.JSON
```

Reply:
This was giving type errors since `MimeType.JSON` is not a string.

Reply:
right, but we (actually just I) have been using the enum in the rest of our code base

Reply:
I see. This one is tricky, let's chat on it tomorrow.